modules/web/src/main/scala/observe/ui/Main.scala

```diff
@@ -5,67 +5,85 @@ package observe.ui
 
 import cats.effect.IO
 import cats.effect.Resource
-import cats.effect.Sync
 import cats.effect.std.Dispatcher
 import cats.effect.unsafe.implicits._
 import cats.syntax.all.*
 import clue.js.WebSocketJSBackend
 import clue.js.WebSocketJSClient
-import clue.websocket.CloseParams
 import clue.websocket.ReconnectionStrategy
+import crystal.react.*
 import crystal.react.hooks.*
-import io.circe.Json
-import io.circe.syntax.given
 import japgolly.scalajs.react.*
 import japgolly.scalajs.react.extra.router._
 import japgolly.scalajs.react.vdom.html_<^.*
 import log4cats.loglevel.LogLevelLogger
+import lucuma.core.model.StandardRole
+import lucuma.core.model.StandardUser
 import lucuma.schemas.ObservationDB
 import lucuma.ui.enums.Theme
+import lucuma.ui.sso.SSOClient
+import lucuma.ui.sso.UserVault
 import observe.ui.model.RootModel
+import org.http4s.circe.*
+import org.http4s.dom.FetchClientBuilder
+import org.http4s.syntax.all.*
 import org.scalajs.dom
 import org.scalajs.dom.Element
 import org.typelevel.log4cats.Logger
 import typings.loglevel.mod.LogLevelDesc
 
 import java.util.concurrent.TimeUnit
 import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.*
 import scala.scalajs.js.annotation.JSExport
 import scala.scalajs.js.annotation.JSExportTopLevel
 
 @JSExportTopLevel("Main")
-object Main {
+object Main:
 
   @JSExport
   def runIOApp(): Unit = run.unsafeRunAndForget()
 
-  def setupLogger[F[_]: Sync](level: LogLevelDesc): F[Logger[F]] = Sync[F].delay {
+  private def setupLogger(level: LogLevelDesc): IO[Logger[IO]] = IO:
     LogLevelLogger.setLevel(level)
-    LogLevelLogger.createForRoot[F]
-  }
+    LogLevelLogger.createForRoot[IO]
 
-  def setupDOM[F[_]: Sync]: F[Element] = Sync[F].delay(
-    Option(dom.document.getElementById("root")).getOrElse {
+  private val setupDOM: IO[Element] = IO:
+    Option(dom.document.getElementById("root")).getOrElse:
       val elem = dom.document.createElement("div")
       elem.id = "root"
       dom.document.body.appendChild(elem)
       elem
-    }
-  )
 
-  val (router, routerCtl) =
+  private val (router, _ /*routerCtl*/ ) =
     RouterWithProps.componentAndCtl(BaseUrl.fromWindowOrigin, Routing.config)
 
-  val mainApp =
+  private val mainApp =
     ScalaFnComponent
-      .withHooks[Unit]
-      .useStateView(RootModel.Initial)
-      .render((_, state) => router(state))
+      .withHooks[Either[Throwable, Option[UserVault]]]
```

Comment:
No `Props`?

Reply:
It's such a small component that it didn't seem necessary, but we can extract it into another file with a `case class` for props if need be.